### PR TITLE
update package.json

### PR DESCRIPTION
**Pull request by priley86**
* updates package.json with dependencies (so npm published distrubution is accurate) for [PTNFLY-1626](https://patternfly.atlassian.net/browse/PTNFLY-1626)
* add `.npmignore` to allow `dist` files in npm distribution
* add index.js to allow `Browserify` or CommonJS support. Noted [here](http://blog.npmjs.org/post/114584444410/using-angulars-new-improved-browserify-support)

### CODE REVIEW

**Comment by priley86** [comment]
@dgutride @dlabrecq @jeff-phillips-18 please take a look

File: package.json [hunk 1]
```diff
@@ -4,8 +4,17 @@
...
+  "keywords": ["angular", "patternfly"],
+  "main": "index.js",
   "homepage": "https://github.com/patternfly/angular-patternfly",
-  "dependencies": {},
+  "dependencies": {
+    "angular": "1.3.0 - 1.5.*",
+    "angular-animate": "1.3.0 - 1.5.*",
+    "angular-sanitize": "1.3.0 - 1.5.*",
+    "angular-ui-bootstrap": "0.13.x",
+    "lodash": "3.x",
```
Comment (dtaylor113) [inline package.json]: Hmm...there was some talk a while ago about consumers not wanting to be required to use lodash.  At the time we coded some functions to test to see if lodash was installed, if not, do the function in plain angularjs.

Comment (priley86) [inline package.json]: @dtaylor113 i'm not positive, but I believe angular-p.f. still depends on Lodash. @dlabrecq can you confirm?

**Comment by jeff-phillips-18** [comment]
Agreed, the strategy was to not require lodash. Hopefully nothing was added to make it a requirement.

File: package.json (hunk 1 above)
Comment (dlabrecq) [inline package.json]: I see lodash in bower.json.

**Comment by priley86** [comment]
i've updated `index.js` and added a simple example of using angular-patternfly with Browserify [here](https://github.com/priley86/angular-browserify)

File: package.json (hunk 1 above)
Comment (priley86) [inline package.json]: It appears Lodash is still in use. I see references to `._remove`, `_.findWhere` and `_.find` which are not scoped like the references to `_.merge` which check for `angular.merge` first.

I think we should address this in a separate PR.

Comment (dtaylor113) [inline package.json]: @priley86 Ok on addressing that in a separate PR.  Please create a JIRA story for this issue.

Comment (priley86) [inline package.json]: Sure! I've created [PTNFLY-1711](https://patternfly.atlassian.net/browse/PTNFLY-1711) to address this. Let me know if this works.

**Comment by priley86** [comment]
any more thoughts here ?

**Comment by jeff-phillips-18** [comment]
I'm OK with this then.

**Comment by priley86** [comment]
@jeff-phillips-18 @dtaylor113  I'm happy to make these updates for lodash in a future PR. Just didn't want to conflate it.

**Comment by priley86** [comment]
@jeff-phillips-18 @dtaylor113 ok to merge?

**Comment by dtaylor113** [comment]
I'm ok with this as well.  The lodash requirement would only be for the npm only install, correct?  If so, then this won't break any existing customers ('cause angular-pf doesn't have an npm yet :-)

**Pull request merged**